### PR TITLE
add note about `remove_duplicate_unreachable_blocks` in source comments

### PR DESCRIPTION
**Pull request by miguelraz**
gotta write down negative results, for science!

@Nilstrieb 

https://github.com/rust-lang/rust/pull/110524#issuecomment-1515132926

Please signal for rollup.

### CODE REVIEW

**Comment by compiler-errors** [comment]
I don't think we usually comment about failed perf attempts, since I would assume very hot code (e.g. trait selection) would be riddled with these comments -- though perhaps someone like @nnethercote knows about best practice here.

File: compiler/rustc_mir_transform/src/simplify.rs [hunk 1]
```diff
@@ -290,6 +290,8 @@ pub fn combine_duplicate_switch_targets(terminator: &mut Terminator<'_>) {
     }
 }
 
+// Note: Trying to Turn `duplicates` into a `SmallVec` for performance did not bear fruit, see
+// https://github.com/rust-lang/rust/pull/110524
```
Comment (compiler-errors) [inline compiler/rustc_mir_transform/src/simplify.rs:294]: Even if we do want to leave a note here, it should probably left as a comment on the field instead?

Comment (nnethercote) [reply compiler/rustc_mir_transform/src/simplify.rs:294]: I agree.

**Comment by miguelraz** [comment]
That sounds reasonable - I'm open to whatever maintainers feel is best.

**Comment by nnethercote** [comment]
Comments about perf are good when the code has unexpected things in it. E.g. often code that is optimized is written in a way that is different to the obvious way, and a comment about that is helpful: "this code is hot, hence...".

This PR's comment is effectively saying "this code is written in the obvious way, not an optimized way". And @compiler-errors is right that we don't normally write comments like that. If there were many of them I think they'd get annoying. If there was a reason to think that this code would be especially hot, then maybe, but it doesn't strike me that way.

**Comment by Nilstrieb** [comment]
I just thought that it'd be quite likely that someone in the future might see this and see an opportunity to optimize. Only to waste their time. But if you think it's not worth it, fair.

**Comment by apiraino** [comment]
Switching to waiting on author, IIUC indications were left for some changes? thanks

@rustbot author

**Comment by compiler-errors** [comment]
I'm gonna close this since it doesn't seem like adding this note is useful. Thanks for the PR!

**Pull request closed without merging**